tests/test_autonomy/test_cli/test_scaffold_fsm.py

```diff
@@ -68,21 +69,31 @@ def setup_class(cls) -> None:
     def test_run(self, fsm_spec_file: Path) -> None:
         """Test run."""
 
-        my_skill = f"test_{fsm_spec_file.parts[-2]}"
+        my_skill = f"{fsm_spec_file.parts[-2]}"
         self.cli_options[-3] = my_skill
         path_to_spec_file = Path(ROOT_DIR) / fsm_spec_file
         args = [*self.cli_options, path_to_spec_file]
         result = self.run_cli_command(*args, cwd=self._get_cwd())
         assert result.exit_code == 0
 
+    def test_autonomy_test_on_scaffolded_fsm(self, fsm_spec_file: Path):
```

Comment:
This test class makes assumptions on the order in which tests are executed. That should be avoided. Please refactor so this is not required.

Further, the problem with the temp files mentioned is also related to this.

Reply:
I added `pytest.mark.order` to address this. The alternative would require one to separate into different test classes and `setup_class`, the scaffolding and `teardown_class` would then be repeatedly run for each of the tests